src/System.Private.ServiceModel/src/System/IdentityModel/CryptoHelper.cs

```diff
@@ -61,7 +97,305 @@ internal static bool IsAsymmetricAlgorithm(string algorithm)
+                        }
+
+                        if (algorithmObject == null)
+                        {
+                            s_algorithmDelegateDictionary[algorithm] = null;
+                        }
+                        else
+                        {
+                            defaultObject = GetDefaultAlgorithm(algorithm);
+                            if (defaultObject != null && defaultObject.GetType() == algorithmObject.GetType())
+                            {
+                                s_algorithmDelegateDictionary[algorithm] = null;
+                            }
+                            else
+                            {
+                                // Create a factory delegate which returns new instances of the algorithm type for later calls.
+                                Type algorithmType = algorithmObject.GetType();
+                                System.Linq.Expressions.NewExpression algorithmCreationExpression = Linq.Expressions.Expression.New(algorithmType);
+                                System.Linq.Expressions.LambdaExpression creationFunction = Linq.Expressions.Expression.Lambda<Func<object>>(algorithmCreationExpression);
+                                delegateFunction = creationFunction.Compile() as Func<object>;
+
+                                if (delegateFunction != null)
+                                {
+                                    s_algorithmDelegateDictionary[algorithm] = delegateFunction;
+                                }
+                                return algorithmObject;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                if (delegateFunction != null)
+                {
+                    return delegateFunction.Invoke();
+                }
+            }
+
+            //
+            // This is a fallback in case CryptoConfig fails to return a valid
+            // algorithm object. CrytoConfig does not understand all the uri's and
+            // can return a null in that case, in which case it is our responsibility
+            // to fallback and create the right algorithm if it is a uri we understand
+            //
+            switch (algorithm)
+            {
+                case SHA256String:
+                case SecurityAlgorithms.Sha256Digest:
+                    return SHA256.Create();
+                case SecurityAlgorithms.Sha1Digest:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return SHA1.Create();
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SecurityAlgorithms.HmacSha1Signature:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return new HMACSHA1(GenerateRandomBytes(64));
```

Comment:
The earlier invocation was 64 bytes.  This is, I think, 64 bits?

Reply:
I'll switch the the parameterless constructor

---
In reply to: [274005077](https://github.com/dotnet/wcf/pull/3538#discussion_r274005077) [](ancestors = 274005077)